src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -0,0 +1,4954 @@
+ *	int i;
+ *
+ *	for (i = 0; i < MAX_FD_DUP2ED; i++)	{
+ *		if (fd_dest == fd_dup2_list[i].fd_dest)	{
+ *			return fd_dup2_list[i].fd_src;
+ *		}
+ *	}
+ *	return -1;
+ * }
+ */
+
+static int
+check_path_with_dirfd(int dirfd, char *full_path_out, const char *rel_path, int *error)
+{
+	int len_str;
+
+	*error = 0;
+
+	if (dirfd >= FD_DIR_BASE) {
+		len_str = snprintf(full_path_out, DFS_MAX_PATH, "%s/%s",
+				   dir_list[dirfd - FD_DIR_BASE]->path, rel_path);
+		if (len_str >= DFS_MAX_PATH)
+			goto out_toolong;
+	} else if (dirfd == AT_FDCWD) {
+		len_str = snprintf(full_path_out, DFS_MAX_PATH, "%s/%s", cur_dir, rel_path);
+		if (len_str >= DFS_MAX_PATH)
+			goto out_toolong;
+	} else {
+		char    path_fd_dir[32];
+		ssize_t bytes_read;
+
+		/* 32 bytes should be more than enough */
+		snprintf(path_fd_dir, 32, "/proc/self/fd/%d", dirfd);
+		bytes_read = readlink(path_fd_dir, full_path_out, DFS_MAX_PATH);
+		if (bytes_read >= DFS_MAX_PATH) {
+			full_path_out[DFS_MAX_PATH - 1] = 0;
+			goto out_toolong;
+		} else if (bytes_read < 0) {
+			if (bLog)
+				D_ERROR("readlink() failed in check_path_with_dirfd(%d). %s",
+					dirfd, strerror(errno));
+			*error = errno;
+		}
+		len_str = snprintf(full_path_out + bytes_read, DFS_MAX_PATH - bytes_read, "/%s",
+				   rel_path);
+		if ((len_str + (int)bytes_read) >= DFS_MAX_PATH)
+			goto out_toolong;
+		full_path_out[len_str + (int)bytes_read] = 0;
+	}
+
+	return query_dfs_mount(full_path_out);
+
+out_toolong:
+	if (bLog)
+		D_ERROR("path is too long");
+	*error = ENAMETOOLONG;
+	return (-1);
+}
+
+static int
+open_common(int (*real_open)(const char *pathname, int oflags, ...), const char *caller_name,
+	    const char *pathname, int oflags, ...)
+{
+	unsigned int     mode     = 0664;
+	int              two_args = 1, rc, is_target_path, idx_fd, idx_dirfd;
+	dfs_obj_t       *dfs_obj;
+	dfs_obj_t       *parent;
+	mode_t           mode_query = 0, mode_parent = 0;
+	struct dfs_mt   *dfs_mt;
+	char             item_name[DFS_MAX_NAME];
+	char             parent_dir[DFS_MAX_PATH];
+	char             full_path[DFS_MAX_PATH];
+
+	if (pathname == NULL) {
+		errno = EFAULT;
+		return (-1);
+	}
+	if (oflags & O_CREAT) {
+		va_list arg;
+
+		va_start(arg, oflags);
+		mode = va_arg(arg, unsigned int);
+		mode = mode & mode_not_umask;
+		va_end(arg);
+		two_args = 0;
+	}
+
+	if (!hook_enabled)
+		goto org_func;
+
+	rc = parse_path(pathname, &is_target_path, &parent, item_name, parent_dir,
+			full_path, &dfs_mt);
+	if (rc != 0 && errno == ENOENT)
+		D_GOTO(out_error, rc = ENOENT);
+
+	if (!is_target_path)
+		goto org_func;
+
+	if (oflags & __O_TMPFILE) {
+		if (!parent && (strncmp(item_name, "/", 2) == 0))
+			rc = dfs_access(dfs_mt->dfs, NULL, NULL, X_OK | W_OK);
+		else
+			rc = dfs_access(dfs_mt->dfs, parent, item_name, X_OK | W_OK);
+		if (rc) {
+			if (rc == 1)
+				rc = 13;
+			D_GOTO(out_error, rc);
+		}
+	}
+
+	if ((oflags & O_RDWR) && (oflags & O_CREAT)) {
+		mode_parent = dfs_obj_mode(dfs_mt->dfs, parent);
```

Comment:
i don't think this is necessary either as the mode is already obtained from:
https://github.com/daos-stack/daos/pull/11345/files#diff-5bf38ce94f48630a1e3e81661870ed9bce01de6715bd10ab9195c4677263d518R863
?

Reply:
Thank you! Fixed as you suggested.